cmd/system-probe/probe.go

```diff
@@ -160,21 +163,17 @@ func getClientID(req *http.Request) string {
 	return clientID
 }
 
-func writeConnections(w http.ResponseWriter, cs *ebpf.Connections) {
-	jw := &jwriter.Writer{}
```

Comment:
I'm afraid I'm introducing here a regression by getting rid of this recent `jwriter` change. Any thoughts how I should approach this buffer pool thing with multiple marshalers? Should I change the `Marshal` interface to something that accepts a buffer as an argument? Or create a some sort of `Writer` object that manages this pool of buffers as in the `easyjson` library?

Reply:
Could you change the ebpf.Marshaler interface to take a `writer` instead of returning a `[]byte` ? That way you can lift this code into `jSerializer` and keep all the performance gains.

It's probably okay to generate a `[]byte` for the protobuf because it'll be so much smaller than the json version.

Reply:
I like the idea. I might be able to achieve by using https://godoc.org/github.com/golang/protobuf/proto#Buffer